pkg/kubelet/kubelet.go

```diff
@@ -1957,8 +1962,14 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	// Use WithoutCancel instead of a new context.TODO() to propagate trace context
 	// Call the container runtime's SyncPod callback
 	sctx := context.WithoutCancel(ctx)
+	syncStartTime := kl.clock.Now()
 	result := kl.containerRuntime.SyncPod(sctx, pod, podStatus, pullSecrets, kl.backOff)
 	kl.reasonCache.Update(pod.UID, result)
+	if utilfeature.DefaultFeatureGate.Enabled(features.EventedPLEG) {
+		// Wait for PLEG to get events from all containers which are started or stopped in syncPod().
+		// This should be done here because what the worker requested remains in PodSyncResult.
+		kl.waitEventsBySyncPod(pod, &result, syncStartTime)
```

Comment:
so if waiting finished, there is no other change following this. What am I missing?

Reply:
With this fix, a pod worker will be waiting here in its main loop:
https://github.com/kubernetes/kubernetes/blob/5fc4e71a306b419b1cf5be81374f2435ecd9ccc7/pkg/kubelet/pod_workers.go#L1286

Then, the worker will not need to wait at `GetNewerThan()` before next sync because the containers are already in the desired state:
https://github.com/kubernetes/kubernetes/blob/5fc4e71a306b419b1cf5be81374f2435ecd9ccc7/pkg/kubelet/pod_workers.go#L1256
So, this PR is trying to replace `GetNewerThan()` with `Get()` at `pod_workers.go`.